src/argus_htmx/incidents/views.py

```diff
@@ -66,143 +67,37 @@ def incident_row(request, pk: int):
 
 def incident_detail(request, pk: int):
     incident = get_object_or_404(Incident, id=pk)
-    action_endpoints = {
-        "ack": reverse("htmx:incident-detail-add-ack", kwargs={"pk": pk}),
-        "close": reverse("htmx:incident-detail-close", kwargs={"pk": pk}),
-        "reopen": reverse("htmx:incident-detail-reopen", kwargs={"pk": pk}),
-        "edit_ticket": reverse("htmx:incident-detail-edit-ticket", kwargs={"pk": pk}),
-        "add_ticket": reverse("htmx:incident-detail-add-ticket", kwargs={"pk": pk}),
-    }
     context = {
         "incident": incident,
-        "endpoints": action_endpoints,
         "page_title": str(incident),
     }
     return render(request, "htmx/incidents/incident_detail.html", context=context)
 
 
-def _incident_add_ack(pk: int, formdata, user: User, group: Optional[str] = None):
-    incident = get_object_or_404(Incident, id=pk)
-    is_group_member = None
-    if group:
-        group = get_object_or_404(Group, name=group)
-        is_group_member = user.groups.filter(pk=group.pk).exists()
-    form = AckForm()
-    if formdata:
-        if group and not is_group_member:
-            raise PermissionDenied("User {request.user} is not a member of the correct group")
-        form = AckForm(formdata)
-        if form.is_valid():
-            incident.create_ack(
-                user,
-                description=form.cleaned_data["description"],
-                expiration=form.cleaned_data["expiration"],
-            )
-    context = {
-        "form": form,
-        "incident": incident,
-        "page_title": str(incident),
-        "group": group,
-        "is_group_member": is_group_member,
-    }
-    return incident, context
-
-
-def incident_add_ack(request, pk: int, group: Optional[str] = None):
-    formdata = request.POST or None
-    _, context = _incident_add_ack(pk, formdata, request.user, group)
-    return render(request, "htmx/incidents/incident_add_ack.html", context=context)
-
-
-@require_POST
-def incident_detail_add_ack(request, pk: int, group: Optional[str] = None):
-    formdata = request.POST or None
-    _incident_add_ack(pk, formdata, request.user, group)
-    return redirect("htmx:incident-detail", pk=pk)
-
-
 def get_form_data(request, formclass: forms.Form):
     formdata = request.POST or None
-    incident_ids = []
     cleaned_form = None
     if formdata:
-        incident_ids = request.POST.getlist("selected_incidents", [])
         form = formclass(formdata)
         if form.is_valid():
             cleaned_form = form.cleaned_data
-    return cleaned_form, incident_ids
+    return cleaned_form
 
 
 @require_POST
-def incidents_update(request: HtmxHttpRequest):
+def incidents_update(request: HtmxHttpRequest, pk: Optional[int] = None):
```

Comment:
Optional arguments to a view is a smell.

Reply:
If I may ask, why is that a smell? I quite like that pattern in general.